experiment/logviewer/gcsreader/parsing.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"bufio"
+	"io"
+	"regexp"
+	"strings"
+
+	"github.com/kubernetes/klog"
+)
+
+func processLines(reader io.Reader, regex *regexp.Regexp) ([]*logEntry, error) {
+	var result []*logEntry
+	r := bufio.NewReader(reader)
+	for {
+		line, err := r.ReadBytes('\n')
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, err
+		}
+		matched, entry, err := processLine(line, regex)
+		if err != nil {
+			// TODO There is a problem that files finish with incomplete line
+			klog.Errorf("%s error parsing line %s", err, line)
+		}
+		if matched && err == nil {
```

Comment:
Usually, in go if error occurs we avoid adding else to correct case.

In your code you are writing if matched **&& err == nil**.

I suggest simply adding continue to if err != nil {

klog.Errorf("%s error parsing line %s", err, line)
continue}

And change this to
if matched {

}